include/crocoddyl/multibody/contacts/contact-2d.hxx

```diff
@@ -0,0 +1,127 @@
+  pinocchio::updateFramePlacement(*state_->get_pinocchio().get(), *d->pinocchio, xref_.id);
+  d->v = pinocchio::getFrameVelocity(*state_->get_pinocchio().get(), *d->pinocchio, xref_.id, pinocchio::LOCAL);
+  const Vector3s& vw = d->v.angular();
+  const Vector3s& vv = d->v.linear();
+
+  pinocchio::getFrameJacobian(*state_->get_pinocchio().get(), *d->pinocchio, xref_.id, pinocchio::LOCAL, d->fJf);
+  d->Jc.row(0) = d->fJf.row(0);
+  d->Jc.row(1) = d->fJf.row(2);
+
+  d->a = pinocchio::getFrameAcceleration(*state_->get_pinocchio().get(), *d->pinocchio, xref_.id);
+  d->a0[0] = d->a.linear()[0] + vw[1] * vv[2] - vw[2] * vv[1];
+  d->a0[1] = d->a.linear()[1] + vw[0] * vv[1] - vw[1] * vv[0];
+
+  if (gains_[0] != 0.) {
+    d->a0[0] += gains_[0] * (d->pinocchio->oMf[xref_.id].translation()[0] - xref_.translation[0]);
+    d->a0[1] += gains_[0] * (d->pinocchio->oMf[xref_.id].translation()[2] - xref_.translation[2]);
+  }
+  if (gains_[1] != 0.) {
+    d->a0[0] += gains_[1] * vv[0];
+    d->a0[1] += gains_[1] * vv[2];
+  }
+}
+
+template <typename Scalar>
+void ContactModel2DTpl<Scalar>::calcDiff(const boost::shared_ptr<ContactDataAbstract>& data,
+                                         const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  pinocchio::getJointAccelerationDerivatives(*state_->get_pinocchio().get(), *d->pinocchio, d->joint, pinocchio::LOCAL,
+                                             d->v_partial_dq, d->a_partial_dq, d->a_partial_dv, d->a_partial_da);
+  const std::size_t& nv = state_->get_nv();
+  pinocchio::skew(d->vv, d->vv_skew);
+  pinocchio::skew(d->vw, d->vw_skew);
+  d->fXjdv_dq.noalias() = d->fXj * d->v_partial_dq;
+  d->fXjda_dq.noalias() = d->fXj * d->a_partial_dq;
+  d->fXjda_dv.noalias() = d->fXj * d->a_partial_dv;
+
+  d->da0_dx.leftCols(nv).row(0) = d->fXjda_dq.row(0);
+  d->da0_dx.leftCols(nv).row(0).noalias() += d->vw_skew.row(0) * d->fXjdv_dq.template topRows<3>();
+  d->da0_dx.leftCols(nv).row(0).noalias() -= d->vv_skew.row(0) * d->fXjdv_dq.template bottomRows<3>();
+
+  d->da0_dx.leftCols(nv).row(1) = d->fXjda_dq.row(2);
+  d->da0_dx.leftCols(nv).row(1).noalias() += d->vw_skew.row(2) * d->fXjdv_dq.template topRows<3>();
+  d->da0_dx.leftCols(nv).row(1).noalias() -= d->vv_skew.row(2) * d->fXjdv_dq.template bottomRows<3>();
+
+  d->da0_dx.rightCols(nv).row(0) = d->fXjda_dv.row(0);
+  d->da0_dx.rightCols(nv).row(0).noalias() += d->vw_skew.row(0) * d->fJf.template topRows<3>();
+  d->da0_dx.rightCols(nv).row(0).noalias() -= d->vv_skew.row(0) * d->fJf.template bottomRows<3>();
+
+  d->da0_dx.rightCols(nv).row(1) = d->fXjda_dv.row(2);
+  d->da0_dx.rightCols(nv).row(1).noalias() += d->vw_skew.row(2) * d->fJf.template topRows<3>();
+  d->da0_dx.rightCols(nv).row(1).noalias() -= d->vv_skew.row(2) * d->fJf.template bottomRows<3>();
+
+  if (gains_[0] != 0.) {
+    d->oRf = d->pinocchio->oMf[xref_.id].rotation();
+    typename MathBase::Matrix2s oRf2D;
```

Comment:
I have an error in compilation due to this line: no Matrix2s in MathBase, so I use MatrixXd instead

Reply:
There is no problem in compilation anymore. You were missing a typename before, and now it is fixed.
Try to compile the current branch, and let me know if you have any issues